x-pack/plugins/fleet/common/types/models/agent_policy.ts

```diff
@@ -61,13 +62,7 @@ export interface FullAgentPolicyInput {
 }
 
 export interface FullAgentPolicyOutputPermissions {
-  [role: string]: {
-    cluster: string[];
-    indices: Array<{
-      names: string[];
-      privileges: string[];
-    }>;
-  };
+  [role: string]: PackagePermissions;
```

Comment:
I wonder if the permissions will be per package or per integration?

Reply:
aren't they the same?

Reply:
A package can contain a list of integrations. Today it is only one but very soon it is more then one. That might explain it.

Reply:
Is it documented anywhere how such a package will look like? Right now, packages and integrations are 100% interchangeable, at least as far EPM code is concerned.

Reply:
@mtojek @jen-huang Can you link to the related issues / PR's?

Reply:
Here is an example of a package that ships multiple integrations: https://github.com/elastic/integrations/pull/767

In summary, this is done by a package shipping multiple `policy_templates`. Today all packages only ship one policy template. I am actively working on supporting multiple templates in https://github.com/elastic/kibana/issues/93315. You can imagine the end result being that the `aws` package ships integrations like `AWS EC2`, `AWS DynamoDB`, etc, which show up as separate tiles in our Integrations UI. However, multiple integrations doesn't change our concept of "package policies", a single `aws` package policy can be used to configure all of its available integrations.

Whether a package ships multiple integrations or not has no bearing on the resulting agent YAML, which is the code that is touched in this PR. The agent yaml is only concerned with inputs, input streams, and outputs. So I think it's more of a question of what to call this permissions data structure instead of `PackagePermissions` (and maybe rename `getPackagePermissions` too) to decouple it from the concept of packages/integrations. I'll leave suggestions in other review comments.

Reply:
Are `data_stream`s nested under `policy_templates` or under the package itself? If it's the latter then the code is valid in the sense that it returns the permissions for a _package_, but my understanding is that we don't necessarily want that, so we need to re-estate the #64634 issue